src/coreclr/jit/gentree.cpp

```diff
@@ -12745,55 +12717,6 @@ GenTree* Compiler::gtFoldExpr(GenTree* tree)
 
             return gtFoldExprCompare(tree);
         }
-        else if (op2->OperGet() == GT_COLON)
```

Comment:
Why this block was removed?

Reply:
The idea is that, as we no longer use QMARKs widely, what this block does is not useful.

We create QMARKs of very specific shapes in the compiler, and it would be very weird of we started creating ones with identical `ThenClause`s and `ElseClause`s.

I note that the useful optimization of when the condition is known is still there.